sga/views.py

```diff
@@ -1,41 +1,123 @@
-"""
-sga views
-"""
-import json
-
+from datetime import datetime
 from django.conf import settings
-from django.contrib.staticfiles.templatetags.staticfiles import static
-from django.shortcuts import render
+from django.contrib.auth import login, authenticate
+from django.core.urlresolvers import reverse
+from django.http import Http404, HttpResponseForbidden
+from django.shortcuts import render, redirect
+from django.views.decorators.csrf import csrf_exempt
+
+from sga.backend.authentication import allowed_roles
+from sga.backend.downloads import serve_zip_file
+from sga.backend.constants import Roles, GRADER_TO_STUDENT_CONFIRM, STUDENT_TO_GRADER_CONFIRM, UNASSIGN_GRADER_CONFIRM, \
+    UNASSIGN_STUDENT_CONFIRM
+from sga.forms import StudentAssignmentSubmissionForm, GraderAssignmentSubmissionForm
+from sga.models import Assignment, Submission, Course, Grader, Student, User
 
 
-def get_bundle_url(request, bundle_name):
+@csrf_exempt
+def index(request):
     """
-    Create a URL for the webpack bundle.
+    The index view. Display available programs
     """
-    if settings.DEBUG and settings.USE_WEBPACK_DEV_SERVER:
-        host = request.get_host().split(":")[0]
+    course = Course.objects.first()
+    assignments = course.assignments.all()
+    users = User.objects.all()
+    students = Student.objects.all()
+    graders = Grader.objects.all()
+    admins = course.administrators.all()
+    if "role" not in request:
+        request.role = None
+    return render(request, "sga/index.html", context={
+        "course": course,
+        "assignments": assignments,
+        "users": users,
+        "students": students,
+        "graders": graders,
+        "admins": admins
+    })
```

Comment:
even if this is technically valid code, the readability is not that great,
so when you need to call functions with parameters on multiple lines, please put one parameter per line